test/telemetry.spec.ts

```diff
@@ -124,6 +125,66 @@ describe('Client Telemetry', () => {
       }
     );
   });
+  it('Sends usage telemetry when registered', (done) => {
+    let numRequests = 0;
+
+    createTestServer(
+      (req, res) => {
+        numRequests += 1;
+
+        const telemetry = req.headers['x-stripe-client-telemetry'];
+
+        switch (numRequests) {
+          case 1:
+            expect(telemetry).to.not.exist;
+            break;
+          case 2:
+            expect(telemetry).to.exist;
+            expect(
+              JSON.parse(telemetry).last_request_metrics.request_id
+            ).to.equal('req_1');
+            expect(
+              JSON.parse(telemetry).last_request_metrics.usage
+            ).to.deep.equal(['llama', 'bufo' ]);
+            break;
+          default:
+            expect.fail(`Should not have reached request ${numRequests}`);
+        }
+
+        res.setHeader('Request-Id', `req_${numRequests}`);
+        res.writeHead(200, {'Content-Type': 'application/json'});
+        res.end('{}');
+      },
+      (host, port) => {
+        const stripe = require('../src/stripe.cjs.node.js')(
+          'sk_test_FEiILxKZwnmmocJDUjUNO6pa',
+          {
+            telemetry: true,
+            host,
+            port,
+            protocol: 'http',
+          }
+        );
+
+        const resource = new (StripeResource.extend({
+          boop: StripeResource.method({
+            method: 'POST',
+            fullPath: '/v1/widgets/{widget}/boop',
+            usage: ['llama', 'bufo' ],
+          })}))(stripe)
+
+        return Promise.resolve()
```

Comment:
can we use async await here?

Reply:
Not trivially. `createServer` takes a callback so we have to convert the promise to a callback. Doable with like an immediately-invoked async function but I'd rather just be consistent with the other test cases.

Reply:
Why doesn't 
```
   try {
          await resource.boop('w_123');
          await resource.boop('w_123');
          expect(numRequests).to.equal(2);
        } finally {
          done();
        }
``` 

work?

Reply:
It does.

I just async'd the whole file.